jetty-core/jetty-http3/jetty-http3-server/src/main/java/org/eclipse/jetty/http3/server/AbstractHTTP3ServerConnectionFactory.java

```diff
@@ -53,7 +53,10 @@ public AbstractHTTP3ServerConnectionFactory(ServerQuicConfiguration quicConfigur
         http3Configuration.setUseInputDirectByteBuffers(httpConfiguration.isUseInputDirectByteBuffers());
         http3Configuration.setUseOutputDirectByteBuffers(httpConfiguration.isUseOutputDirectByteBuffers());
         http3Configuration.setMaxRequestHeadersSize(httpConfiguration.getRequestHeaderSize());
-        http3Configuration.setMaxResponseHeadersSize(httpConfiguration.getMaxResponseHeaderSize());
+        int maxResponseHeaderSize = httpConfiguration.getMaxResponseHeaderSize();
+        if (maxResponseHeaderSize < 0)
+            maxResponseHeaderSize = getHttpConfiguration().getResponseHeaderSize();
```

Comment:
Are you sure you don't want a convenience method
```suggestion
        int maxResponseHeaderSize = HttpConfiguration.getEffectiveMaxResponseHeaderSize(httpConfiguration);
```

Reply:
@gregw I am thinking that all this backport code should not be merged to 12.1.x, and so I would like to avoid another public method in `HttpConfiguration`.

Reply:
So how do you propose to handle this in 12.1?   Just not have the -1 support?    
I think we should have the -1 support, but change the default to not be -1.